gymnasium/spaces/utils.py

```diff
@@ -271,7 +271,7 @@ def _flatten_oneof(space: OneOf, x: tuple[int, Any]) -> NDArray[Any]:
     max_flatdim = flatdim(space) - 1  # Don't include the index
     if flat_sample.size < max_flatdim:
         padding = np.full(
-            max_flatdim - flat_sample.size, np.nan, dtype=flat_sample.dtype
+            max_flatdim - flat_sample.size, flat_sample[0], dtype=flat_sample.dtype
```

Comment:
I'm now wondering if it might be better to use `np.empty` instead if we're not using a dedicated placeholder value.

Either way, the array will be filled with some throwaway data, but with empty we're not pretending that this data has any actual meaning.

Reply:
I ran a quick check and this fails `test_flat_space_contains_flat_points` as the flattened version might not be contained within the flattened space. 
But this is just a weird artifact of `flatten` rather than anything actually incorrect with the approach